File: BOJ/graph&Traversal/1926.py
import pprint
import sys

sys.setrecursionlimit(10000)



n,m=map(int,input().split())
max=0
tmp=0
def dfs_R(x,y):
    global max
    global tmp
    if x>=n or x <=-1 or y<=-1 or y>=m:
        return False

    if visited[x][y]==False and graph[x][y]==1:
        visited[x][y]=True
        tmp+=1
        dfs_R(x-1,y)
        dfs_R(x,y-1)
        dfs_R(x+1,y)
        dfs_R(x,y+1)
        return True
    return False
    
    

graph=[]
cnt=0

for i in range(n):
    temp=list(map(int,input().split()))
    graph.append(temp)
visited=[[False for i in range(m)]for i in range(n)]

for i in range(n):
    for j in range(m):
       if visited[i][j]==False:
        if(dfs_R(i,j)==True):
            if tmp>max:
                max=tmp
            tmp=0
            cnt+=1
        



print(cnt)
print(max)

File: BOJ/graph&Traversal/1926.c
#include<stdio.h>
#include<stdlib.h>

int max=0;
int tmp=0;
int n,m=0;
int cnt=0;


int dfs_R(int x,int y,int**visit,int**graph){
    if(x>=n||x<=-1||y<=-1||y>=m){
        return 0;
    }

    if(visit[x][y]==0&&graph[x][y]==1){
        visit[x][y]=1;
        tmp+=1;
        dfs_R(x-1,y,visit,graph);
        dfs_R(x,y-1,visit,graph);
        dfs_R(x+1,y,visit,graph);
        dfs_R(x,y+1,visit,graph);
        return 1;
    }
    return 0;
}

int main(){
    
    scanf("%d %d",&n,&m);
    int **graph=malloc(sizeof(int*)*n);
    int **visit=malloc(sizeof(int*)*n);
    for(int i=0; i<n; i++){
        visit[i]=malloc(sizeof(int)*m);
        graph[i]=malloc(sizeof(int)*m);
    }
   
    for(int i=0; i<n; i++){
        for(int j=0; j<m; j++){
            scanf("%d",&graph[i][j]);
            visit[i][j]=0;
        }
       
    }
    

    for(int i=0; i<n; i++){
        for(int j=0; j<m; j++){
            if(visit[i][j]==0){
                if(dfs_R(i,j,visit,graph)==1){
                    if(tmp>max){
                        max=tmp;
                    }
                    tmp=0;
                    cnt+=1;
                }
            }
        }
    }

    printf("%d\n%d",cnt,max);

    


}

